net_transport.go

```diff
@@ -199,6 +205,28 @@ func (n *NetworkTransport) SetHeartbeatHandler(cb func(rpc RPC)) {
 	n.heartbeatFn = cb
 }
 
+// UseStream re-initializes the stream layer
+func (n *NetworkTransport) UseStream(s StreamLayer) {
```

Comment:
Consider making this return an error and do nothing if CloseStream was not previously called to prevent leaking connections due to programmer error. 
One way to do this would be to set a boolean that the stream layer has been closed when it receives an event on the Done channel in the context used here, and check against the state of that boolean before reinitializing the stream layer.